catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -260,6 +269,80 @@ private DeleteResponse performLocalDelete(DeleteRequest deleteRequest,
         return deleteResponse;
     }
 
+    //
+    // Private helper methods
+    //
+
+    private DeleteRequest rewriteRequestToAvoidHistoryConflicts(DeleteRequest deleteRequest)
+            throws UnsupportedQueryException {
+        final String attributeName = deleteRequest.getAttributeName();
+        if (Metacard.ID.equals(attributeName)) {
+            return deleteRequest;
+        }
+
+        Filter filter = deleteRequest.getAttributeValues()
+                .stream()
+                .map(Object::toString)
+                .map((val) -> getNonhistoryFilter(attributeName, val))
+                .collect(Collectors.collectingAndThen(Collectors.toList(),
+                        frameworkProperties.getFilterBuilder()::anyOf));
+
+        SourceResponse response = sourceOperations.getCatalog()
+                .query(new QueryRequestImpl(new QueryImpl(filter,
+                        1,
+                        0,
+                        SortBy.NATURAL_ORDER,
+                        false,
+                        TimeUnit.SECONDS.toMillis(10))));
+
+        if (response.getResults()
```

Comment:
Checking the result size might not catch all cases.  It is possible one or more identifiers do not match but there are duplicates for others so that the final size matches.